From 76c7142bd2595184826722063598fddf62031cca Mon Sep 17 00:00:00 2001
From: Tobias Stoeckmann <tobias@stoeckmann.org>
Date: Sun, 2 Aug 2026 14:31:33 +0200
Subject: [PATCH 3/3] Remove bidder name

The name was only needed to find the appropriate bidder for a filter.
Since this has been resolved by accessing the last bidder, since in these
situations the bidder is added first, then the filter, the name is not
needed anymore.

Upstream-Status: Submitted [https://github.com/libarchive/libarchive/pull/3346]
Signed-off-by: Tobias Stoeckmann <tobias@stoeckmann.org>
Signed-off-by: Sumanth Gavini <sumanth.gavini@windriver.com>
---
 libarchive/archive_read.c                         | 2 --
 libarchive/archive_read_append_filter.c           | 1 -
 libarchive/archive_read_private.h                 | 3 ---
 libarchive/archive_read_support_filter_bzip2.c    | 2 +-
 libarchive/archive_read_support_filter_compress.c | 2 +-
 libarchive/archive_read_support_filter_grzip.c    | 2 +-
 libarchive/archive_read_support_filter_gzip.c     | 2 +-
 libarchive/archive_read_support_filter_lrzip.c    | 2 +-
 libarchive/archive_read_support_filter_lz4.c      | 2 +-
 libarchive/archive_read_support_filter_lzop.c     | 2 +-
 libarchive/archive_read_support_filter_program.c  | 2 +-
 libarchive/archive_read_support_filter_rpm.c      | 2 +-
 libarchive/archive_read_support_filter_uu.c       | 2 +-
 libarchive/archive_read_support_filter_xz.c       | 6 +++---
 libarchive/archive_read_support_filter_zstd.c     | 2 +-
 15 files changed, 14 insertions(+), 20 deletions(-)

diff --git a/libarchive/archive_read.c b/libarchive/archive_read.c
index e844e42d..406d84de 100644
--- a/libarchive/archive_read.c
+++ b/libarchive/archive_read.c
@@ -1290,7 +1290,6 @@ __archive_read_register_format(struct archive_read *a,
 int
 __archive_read_register_bidder(struct archive_read *a,
 	void *bidder_data,
-	const char *name,
 	const struct archive_read_filter_bidder_vtable *vtable)
 {
 	struct archive_read_filter_bidder *bidder;
@@ -1307,7 +1306,6 @@ __archive_read_register_bidder(struct archive_read *a,
 		memset(a->bidders + i, 0, sizeof(a->bidders[0]));
 		bidder = (a->bidders + i);
 		bidder->data = bidder_data;
-		bidder->name = name;
 		bidder->vtable = vtable;
 		if (bidder->vtable->bid == NULL || bidder->vtable->init == NULL) {
 			archive_set_error(&a->archive, ARCHIVE_ERRNO_PROGRAMMER,
diff --git a/libarchive/archive_read_append_filter.c b/libarchive/archive_read_append_filter.c
index 9dd9f536..5cc8ff55 100644
--- a/libarchive/archive_read_append_filter.c
+++ b/libarchive/archive_read_append_filter.c
@@ -174,7 +174,6 @@ archive_read_append_filter_program_signature(struct archive *_a,
     __archive_read_free_filters(a);
     return (ARCHIVE_FATAL);
   }
-  b->name = a->filter->name;
 
   a->bypass_filter_bidding = 1;
   return r;
diff --git a/libarchive/archive_read_private.h b/libarchive/archive_read_private.h
index 3bd8de2d..e30b6c2b 100644
--- a/libarchive/archive_read_private.h
+++ b/libarchive/archive_read_private.h
@@ -68,8 +68,6 @@ struct archive_read_filter_bidder_vtable {
 struct archive_read_filter_bidder {
 	/* Configuration data for the bidder. */
 	void *data;
-	/* Name of the filter */
-	const char *name;
 	const struct archive_read_filter_bidder_vtable *vtable;
 };
 
@@ -250,7 +248,6 @@ int	__archive_read_register_format(struct archive_read *a,
 
 int __archive_read_register_bidder(struct archive_read *a,
 		void *bidder_data,
-		const char *name,
 		const struct archive_read_filter_bidder_vtable *vtable);
 
 const void *__archive_read_ahead(struct archive_read *, size_t, ssize_t *);
diff --git a/libarchive/archive_read_support_filter_bzip2.c b/libarchive/archive_read_support_filter_bzip2.c
index 588953f2..1cfc55dc 100644
--- a/libarchive/archive_read_support_filter_bzip2.c
+++ b/libarchive/archive_read_support_filter_bzip2.c
@@ -91,7 +91,7 @@ archive_read_support_filter_bzip2(struct archive *_a)
 {
 	struct archive_read *a = (struct archive_read *)_a;
 
-	if (__archive_read_register_bidder(a, NULL, "bzip2",
+	if (__archive_read_register_bidder(a, NULL,
 				&bzip2_bidder_vtable) != ARCHIVE_OK)
 		return (ARCHIVE_FATAL);
 
diff --git a/libarchive/archive_read_support_filter_compress.c b/libarchive/archive_read_support_filter_compress.c
index 4e1f1dc5..4763995e 100644
--- a/libarchive/archive_read_support_filter_compress.c
+++ b/libarchive/archive_read_support_filter_compress.c
@@ -160,7 +160,7 @@ archive_read_support_filter_compress(struct archive *_a)
 {
 	struct archive_read *a = (struct archive_read *)_a;
 
-	return __archive_read_register_bidder(a, NULL, "compress (.Z)",
+	return __archive_read_register_bidder(a, NULL,
 			&compress_bidder_vtable);
 }
 
diff --git a/libarchive/archive_read_support_filter_grzip.c b/libarchive/archive_read_support_filter_grzip.c
index 9d9ad46e..8457ec7c 100644
--- a/libarchive/archive_read_support_filter_grzip.c
+++ b/libarchive/archive_read_support_filter_grzip.c
@@ -62,7 +62,7 @@ archive_read_support_filter_grzip(struct archive *_a)
 {
 	struct archive_read *a = (struct archive_read *)_a;
 
-	if (__archive_read_register_bidder(a, NULL, "grzip",
+	if (__archive_read_register_bidder(a, NULL,
 				&grzip_bidder_vtable) != ARCHIVE_OK)
 		return (ARCHIVE_FATAL);
 
diff --git a/libarchive/archive_read_support_filter_gzip.c b/libarchive/archive_read_support_filter_gzip.c
index 89ae0f37..34b174b7 100644
--- a/libarchive/archive_read_support_filter_gzip.c
+++ b/libarchive/archive_read_support_filter_gzip.c
@@ -96,7 +96,7 @@ archive_read_support_filter_gzip(struct archive *_a)
 {
 	struct archive_read *a = (struct archive_read *)_a;
 
-	if (__archive_read_register_bidder(a, NULL, "gzip",
+	if (__archive_read_register_bidder(a, NULL,
 				&gzip_bidder_vtable) != ARCHIVE_OK)
 		return (ARCHIVE_FATAL);
 
diff --git a/libarchive/archive_read_support_filter_lrzip.c b/libarchive/archive_read_support_filter_lrzip.c
index 33910fe1..a1d45d77 100644
--- a/libarchive/archive_read_support_filter_lrzip.c
+++ b/libarchive/archive_read_support_filter_lrzip.c
@@ -61,7 +61,7 @@ archive_read_support_filter_lrzip(struct archive *_a)
 {
 	struct archive_read *a = (struct archive_read *)_a;
 
-	if (__archive_read_register_bidder(a, NULL, "lrzip",
+	if (__archive_read_register_bidder(a, NULL,
 				&lrzip_bidder_vtable) != ARCHIVE_OK)
 		return (ARCHIVE_FATAL);
 
diff --git a/libarchive/archive_read_support_filter_lz4.c b/libarchive/archive_read_support_filter_lz4.c
index 19086a1f..28189a07 100644
--- a/libarchive/archive_read_support_filter_lz4.c
+++ b/libarchive/archive_read_support_filter_lz4.c
@@ -118,7 +118,7 @@ archive_read_support_filter_lz4(struct archive *_a)
 {
 	struct archive_read *a = (struct archive_read *)_a;
 
-	if (__archive_read_register_bidder(a, NULL, "lz4",
+	if (__archive_read_register_bidder(a, NULL,
 				&lz4_bidder_vtable) != ARCHIVE_OK)
 		return (ARCHIVE_FATAL);
 
diff --git a/libarchive/archive_read_support_filter_lzop.c b/libarchive/archive_read_support_filter_lzop.c
index ebfaf01e..5a552fce 100644
--- a/libarchive/archive_read_support_filter_lzop.c
+++ b/libarchive/archive_read_support_filter_lzop.c
@@ -109,7 +109,7 @@ archive_read_support_filter_lzop(struct archive *_a)
 {
 	struct archive_read *a = (struct archive_read *)_a;
 
-	if (__archive_read_register_bidder(a, NULL, "lzop",
+	if (__archive_read_register_bidder(a, NULL,
 				&lzop_bidder_vtable) != ARCHIVE_OK)
 		return (ARCHIVE_FATAL);
 
diff --git a/libarchive/archive_read_support_filter_program.c b/libarchive/archive_read_support_filter_program.c
index cf7c3613..a9e02a32 100644
--- a/libarchive/archive_read_support_filter_program.c
+++ b/libarchive/archive_read_support_filter_program.c
@@ -156,7 +156,7 @@ archive_read_support_filter_program_signature(struct archive *_a,
 		memcpy(state->signature, signature, signature_len);
 	}
 
-	if (__archive_read_register_bidder(a, state, NULL,
+	if (__archive_read_register_bidder(a, state,
 				&program_bidder_vtable) != ARCHIVE_OK) {
 		free_state(state);
 		return (ARCHIVE_FATAL);
diff --git a/libarchive/archive_read_support_filter_rpm.c b/libarchive/archive_read_support_filter_rpm.c
index 01bd1bf7..1cfddf75 100644
--- a/libarchive/archive_read_support_filter_rpm.c
+++ b/libarchive/archive_read_support_filter_rpm.c
@@ -71,7 +71,7 @@ archive_read_support_filter_rpm(struct archive *_a)
 {
 	struct archive_read *a = (struct archive_read *)_a;
 
-	return __archive_read_register_bidder(a, NULL, "rpm",
+	return __archive_read_register_bidder(a, NULL,
 			&rpm_bidder_vtable);
 }
 
diff --git a/libarchive/archive_read_support_filter_uu.c b/libarchive/archive_read_support_filter_uu.c
index c0f6f48e..10d49648 100644
--- a/libarchive/archive_read_support_filter_uu.c
+++ b/libarchive/archive_read_support_filter_uu.c
@@ -92,7 +92,7 @@ archive_read_support_filter_uu(struct archive *_a)
 {
 	struct archive_read *a = (struct archive_read *)_a;
 
-	return __archive_read_register_bidder(a, NULL, "uu",
+	return __archive_read_register_bidder(a, NULL,
 			&uudecode_bidder_vtable);
 }
 
diff --git a/libarchive/archive_read_support_filter_xz.c b/libarchive/archive_read_support_filter_xz.c
index 7c8e0cb5..b00ba543 100644
--- a/libarchive/archive_read_support_filter_xz.c
+++ b/libarchive/archive_read_support_filter_xz.c
@@ -115,7 +115,7 @@ archive_read_support_filter_xz(struct archive *_a)
 {
 	struct archive_read *a = (struct archive_read *)_a;
 
-	if (__archive_read_register_bidder(a, NULL, "xz",
+	if (__archive_read_register_bidder(a, NULL,
 				&xz_bidder_vtable) != ARCHIVE_OK)
 		return (ARCHIVE_FATAL);
 
@@ -147,7 +147,7 @@ archive_read_support_filter_lzma(struct archive *_a)
 {
 	struct archive_read *a = (struct archive_read *)_a;
 
-	if (__archive_read_register_bidder(a, NULL, "lzma",
+	if (__archive_read_register_bidder(a, NULL,
 				&lzma_bidder_vtable) != ARCHIVE_OK)
 		return (ARCHIVE_FATAL);
 
@@ -180,7 +180,7 @@ archive_read_support_filter_lzip(struct archive *_a)
 {
 	struct archive_read *a = (struct archive_read *)_a;
 
-	if (__archive_read_register_bidder(a, NULL, "lzip",
+	if (__archive_read_register_bidder(a, NULL,
 				&lzip_bidder_vtable) != ARCHIVE_OK)
 		return (ARCHIVE_FATAL);
 
diff --git a/libarchive/archive_read_support_filter_zstd.c b/libarchive/archive_read_support_filter_zstd.c
index 12eac04c..163f54b5 100644
--- a/libarchive/archive_read_support_filter_zstd.c
+++ b/libarchive/archive_read_support_filter_zstd.c
@@ -87,7 +87,7 @@ archive_read_support_filter_zstd(struct archive *_a)
 {
 	struct archive_read *a = (struct archive_read *)_a;
 
-	if (__archive_read_register_bidder(a, NULL, "zstd",
+	if (__archive_read_register_bidder(a, NULL,
 				&zstd_bidder_vtable) != ARCHIVE_OK)
 		return (ARCHIVE_FATAL);
 
